test/test.js

```diff
@@ -458,6 +462,45 @@ function checkRefTestResults(browser, id, results) {
   });
 }
 
+function browserTestReportHandler(req, res) {
+  var parsedUrl = url.parse(req.url, true);
+  var pathname = parsedUrl.pathname;
+  if (pathname === '/browserTestReports' ||
+    pathname === '/browserWorkerTestReports') {
+
+    var writableStream = fs.createWriteStream(
+      '../coverage/coverageinfo.json');
+    var body = '';
+
+    req.on('data', function (chunk) {
+      body += chunk;
+    });
+
+    res.on('end', function () {
+      var map = libCoverage.createCoverageMap(JSON.parse(body));
+      var summary = libCoverage.createCoverageSummary();
+      map.merge(summary);
+
+      map.files().forEach(function (f) {
+        var fc = map.fileCoverageFor(f),
+          s = fc.toSummary();
+        summary.merge(s);
+        writableStream.write(summary);
+      });
```

Comment:
After removing the uses of `writableStream`, call `fs.writeFileSync` here, followed by `res.end();`. By doing that, the file writing operation can be seen as atomic, so there is no risk of corrupting the file if two requests are initiated simultaneously.

Reply:
So you mean that `res.end()` should be called inside itself ?

Reply:
I meant `req.end()` (to signal that the request has been handled).

Reply:
The last doubt. 

`req.end` previously contained 
```
 req.on('end', function () {
      writableStream.end();
    });
```
However `writeFileSync` has no such `end` event. So what should `req.on('end')` contain now?